google-cloud-compute-v1/samples/firewall.rb

```diff
@@ -0,0 +1,132 @@
+# frozen_string_literal: true
+
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START compute_firewall_list]
+# [START compute_firewall_create]
+# [START compute_firewall_patch]
+# [START compute_firewall_delete]
+
+require "google/cloud/compute/v1"
+
+# [END compute_firewall_delete]
+# [END compute_firewall_patch]
+# [END compute_firewall_create]
+# [END compute_firewall_list]
+
+require_relative "quickstart"
+
+# [START compute_firewall_list]
+# Return a list of all the firewall rules in specified project. Also prints the
+# list of firewall names and their descriptions.
+#
+# @param [String] project project ID or project number of the project you want to use.
+# @return [Array<::Google::Cloud::Compute::V1::Firewall>]
+#     A list of all firewall rules defined for the given project.
+def list_firewall_rules project:
+  client = ::Google::Cloud::Compute::V1::Firewalls::Rest::Client.new
+  firewalls = client.list project: project
+
+  firewall_list = []
+  firewalls.each do |firewall|
+    puts " - #{firewall.name}: #{firewall.description}"
+    firewall_list << firewall
+  end
+
+  firewall_list
+end
+# [END compute_firewall_list]
+
+
+# [START compute_firewall_create]
+# Creates a simple firewall rule allowing for incoming HTTP and HTTPS access from the entire Internet.
+#
+# @param [String] project project ID or project number of the Cloud project you want to use.
+# @param [String] name: name of the rule that is created.
+# @param network: name of the network the rule will be applied to. Available name formats:
+#         * https://www.googleapis.com/compute/v1/projects/{project_id}/global/networks/{network}
+#         * projects/{project_id}/global/networks/{network}
+#         * global/networks/{network}
+def create_firewall_rule project:, name:, network: "global/networks/default"
+  rule = {
+    name: name,
+    direction: "INGRESS",
```

Comment:
We can use `google.cloud.compute.v1.Firewall.Direction` enum and call `to_s` on the `INGRESS` member of that to showcase that some string constants do come from an enum. Or maybe spell it out in a comment if using an enum like that is unidiomatic.

Reply:
I think it's confusing to show the enum when you can't pass an enum value. I'm not sure about the long-term goal but I think we can just pass a string if it's string. The reference also says it's string. https://googleapis.dev/ruby/google-cloud-compute-v1/latest/Google/Cloud/Compute/V1/Firewall.html#direction-instance_method